daktari/checks/terraform.py

```diff
@@ -33,31 +37,34 @@ def __init__(self, required_version: str = "", use_tfenv: bool = False):
             # Read the required tf-version from tfenv (i.e. .terraform-version files)
             tfenv_version = open(".terraform-version", "r").read().strip()
             logging.debug(f"Terraform version required from .terraform-version file is: {tfenv_version}")
-            self.required_version = tfenv_version
+            self.required_version = f"=={tfenv_version}"
             self.suggestions = {OS.GENERIC: "<cmd>tfenv install</cmd>"}
         else:
-            version_string = f"@{required_version}" if required_version else ""
             self.suggestions = {
-                OS.OS_X: f"<cmd>brew tap hashicorp/tap && hashicorp/tap/terraform{version_string}</cmd>",
+                OS.OS_X: "<cmd>brew tap hashicorp/tap && brew install hashicorp/tap/terraform</cmd>",
                 OS.GENERIC: "Install Terraform: https://learn.hashicorp.com/tutorials/terraform/install-cli",
             }
 
     def check(self) -> CheckResult:
         installed_version = get_terraform_version()
-        return self.validate_required_version(
-            "Terraform", installed_version=installed_version, required_version=self.required_version
+        return self.validate_semver_expression(
+            "Terraform", installed_version, self.required_version, self.recommended_version
         )
 
 
-version_pattern = re.compile("Terraform v([0-9]+.[0-9]+.[0-9]+)")
+version_pattern = re.compile("Terraform v([0-9\\.]+)")
 
 
-def get_terraform_version() -> Optional[str]:
+def get_terraform_version() -> Optional[VersionInfo]:
     raw_version = get_stdout("terraform version")
     if raw_version:
         match = version_pattern.search(raw_version)
         if match:
             version_string = match.group(1)
-            logging.debug(f"Terraform version: {version_string}")
-            return str(version_string)
+            try:
+                version = VersionInfo.parse(version_string)
+            except ValueError:
+                return None
+            logging.debug(f"Terraform version: {version}")
+            return version
```

Comment:
Seems like this pattern crops up a lot, are there plans to refactor this out at some point?

Reply:
Yeah, probably should - good idea. Perhaps a function that wraps `VersionInfo.parse()` and does this try/except. I'll put it in this PR since that's where most of the instances of it were added.